### ISSUE TITLE
Internal Server Error, impossible to debug plugin setup

### ISSUE THREAD

**Issue by AndrewFarley** [mtanda/grafana-google-stackdriver-datasource#36]
When trying to setup this plugin, if it doesn't work perfectly it doesn't give any meaningful feedback neither in the console, Grafana usage logs, or on the Grafana GUI on what is wrong and/or how to fix it.

POST http://localhost:3000/api/tsdb/query 500 (Internal Server Error)
datasource.ts:354 TypeError: Cannot read property '' of undefined

![screen shot 2018-08-13 at 3 09 19 pm](https://user-images.githubusercontent.com/470163/44034365-389ee552-9f0d-11e8-8c2e-e27005455bde.png)

**Comment by maxcleme** [comment]
@AndrewFarley Got exactly the same error, but from a standalone grafana installation, did you manage to find a workaround ?

As far as I can tell it seems that the env variable is not recognize by the Go backend, here what I can find in the logs : 

```
t=2018-08-29T13:37:49+0000 lvl=eror msg="Metric request error" logger=context userId=5 orgId=1 uname=xxxxxxxx@xxx.xxx error="rpc error: code = Unknown desc = google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information."
```

**version** :

```
> grafana-server -v
Version 5.1.4 (commit: a5fe24fcc)
```

**Comment by AndrewFarley** [comment]
@maxcleme No, for one of my installations it works fine... for the other it doesn't.   I can't figure out why, I gave up using the server-dictated credentials, and instead am using the client credentials (but then it works for me only)

**Comment by swapnilpatil1682** [comment]
@AndrewFarley  Can you pl tell what steps did you followed where it worked ?

**Comment by maxcleme** [comment]
After more investigation, I manage to get the environment variable recognize in my standalone installation by defining it in `/etc/default/grafana` (which is loaded by the script in `/etc/init.d/grafana-server`). 

However, even if I pass the **Save & Test** step in the datasource configuration, all queries returns empty response.

In the mean time, everything work well by running it in a Docker container, and using the `-e` argument to define the environment variable.

I used the same service-account file in both env, which is really confusing :s

**Comment by AndrewFarley** [comment]
@swapnilpatil1682 Honestly I don't know why one of them works and the other two doesn't.  They are all using kubernetes, using the helm chart, with the same exact configuration as far as I can tell.   But one works and the others don't.  I suspect one of them works because I gave it a service account with "full" api permissions instead of limiting it to only stackdriver.  I suspect (but have not dived into the code) that this plugin uses some obscure API call unrelated to the stackdriver API to check if it's a valid API key.  That's my best guess.  For my two other installs, I am using the OAuth mechanism which works but only for me (not for anyone using Grafana, unless they also do oauth).

**Comment by maxcleme** [comment]
Update: I manage to make it work, both in standalone installation or in docker container. 

First of all, check your Grafana version, I was using the latest tag in my docker environment, but still in 5.1.4 for my standalone installation. 

Once upgraded to 5.2.4, it works well in my standalone installation.

For the environment variable, I also managed to set it in both env :

**docker**

Use the `-e GOOGLE_APPLICATION_CREDENTIALS=/path/to/service_account.json` when starting the container.

**standalone**

Define the following env variable in `/etc/default/grafana-server` :

`GOOGLE_APPLICATION_CREDENTIALS=/path/to/service_account.json`

Regards.

**Comment by vinnu-ssm** [comment]
Hi All @AndrewFarley  @swapnilpatil1682 

Could you please share complete steps(documentation ) to set up stack driver in Grafana  .

You guys also describe above well but it is very high level . 
It ll be very help full for those guys who are going to set up stack driver in Grafana  and having issue like below . 
![image](https://user-images.githubusercontent.com/45870999/49993497-ccc94380-ffac-11e8-8835-5e569a8a8406.png)

**Comment by vinnu-ssm** [comment]
Before above i was also getting "Failed to initiate Plugin" but even after this i was able to get Datasource in grafana for GoogleStackDriver so accordingly i have given required info and getting "err.data.results undefined"

@mtanda  @AndrewFarley  @swapnilpatil1682    - Please share steps or documentation if any . 

![image](https://user-images.githubusercontent.com/45870999/49994023-62190780-ffae-11e8-9ed8-36624c5f5152.png)

**Comment by mtanda** [comment]
@vinaykr253 Hi, thanks for using my plugin :-)
Unfortunately, this plugin is not maintained now, and I recommend you to use the official plugin.
http://docs.grafana.org/features/datasources/stackdriver/

If you have special reason to use my plugin, I'm willing to help you.
